python/docs/source/guide/aggregation.md

```diff
@@ -1,3 +1,33 @@
 # Aggregation
 
-## Windowing
\ No newline at end of file
+Aggregations are _cumulative_, _grouped_, and _windowed_.
+
+Cumulative
+: They reflect all values up to and including the current time.
+
+Grouped
+: They reflect the values for each entity separately.
+
+Windowed
+: They reflect the values within a specific [window](../reference/windows.md).
+
+While most functions operate on specific points in the Timestream, aggregation functions operate on many points.
+Generally, the result of an aggregation represents the result for each key up to and including the current time.
+Conceptually, this is equivalent to running the aggregation at that point in time.
+
+## Windowing
+Aggregations may be configured to operate in a specific time window by providing a `window` argument.
+If no window is specified, the aggregation is over all rows for the entity, up to and including the current time.
+If a window is provided, the result of an aggregation is the result for that entity in the current window up to and including the current time.
+The current window is often incomplete.
+
+## Repeated Aggregation
```

Comment:
Hm, thought about asking whether we'd consider `nested` instead, but I like repeated I think, especially given the new python semantics.